gordian/repo.py

```diff
@@ -35,29 +36,45 @@ def __init__(self, repo_name, github_api_url=None, branch=None, git=None):
         else:
             self.branch_name = f"refs/heads/{datetime.datetime.now().strftime('%Y-%m-%d-%H%M%S.%f')}"
 
-    def get_files(self):
-        if self.files:
-            return self.files
-
-        contents = self.repo.get_contents("")
-        while contents:
-            file_content = contents.pop(0)
-            if file_content.path == 'version':
-                self.version_file = file_content
-            if file_content.type == 'dir':
-                contents.extend(self.repo.get_contents(file_content.path))
-            else:
-                self.files.append(file_content)
+    def get_objects(self, filename, klass=None):
+        file = self.find_file(filename)
+
+        if file is None:
+            raise FileNotFoundError
+
+        if klass:
+            return klass(file, self)
 
-    def get_file(self, filename):
+        _, ext = os.path.splitext(filename)
+
+        if ext in ('.yaml', '.yml'):
+            return YamlFile(file, self)
+        if ext in ('.json'):
+            return JsonFile(file, self)
+
+    def get_files(self):
+        if not self.files:
+            contents = self._repo.get_contents("")
+            while contents:
+                file_content = contents.pop(0)
+                if file_content.path == 'version':
```

Comment:
Nothing related to this PR, but we should revisit the special treatment that we give to the 'version' file.